### DISCUSSION TITLE
是否可以将format_bbox的结果投影到二维图像

### DISCUSSION THREAD

**Discussion by polarbear122** [open-mmlab/mmdetection3d#2303] [Q&A] [answered]
nuscenes-devkit中提供了一个get 2d box函数，可以将3d box投影到二维图像上。
https://github.com/nutonomy/nuscenes-devkit/blob/9b209638ef3dee6d0cdc5ac700c493747f5b35fe/python-sdk/nuscenes/scripts/export_2d_annotations_as_json.py#L98，
其中使用的3d box如下
https://github.com/nutonomy/nuscenes-devkit/blob/9b209638ef3dee6d0cdc5ac700c493747f5b35fe/python-sdk/nuscenes/utils/data_classes.py#L457。
![image](https://user-images.githubusercontent.com/53006681/220707616-93742617-ba64-4939-8521-f2a6afd2e8a8.png)

在mmdet3d中，通过format_bbox函数，我已经获得3d的box，格式如下，
https://github.com/open-mmlab/mmdetection3d/blob/c9541b0db89498fdea5cafd05b7b17f7b625b858/mmdet3d/datasets/nuscenes_dataset.py#L348
![image](https://user-images.githubusercontent.com/53006681/220707161-6188f9ce-8701-4151-8b21-2f590d7299f0.png)

请问是否存在将format bbox函数生成的box直接投影至二维平面的方法？或者转换为nuscenes-devkit中的3d box的方法。

**Comment by JingweiZhang12** [comment, marked as answer]
https://github.com/open-mmlab/mmdetection3d/blob/dev-1.x/mmdet3d/visualization/local_visualizer.py#L442

**Comment by polarbear122** [comment]
Thank you! It will be very helpful for me.
